src/NetworkMap/Certificates/Update.elm

```diff
@@ -16,20 +16,101 @@ limitations under the License.
 
 -}
 
-import Array
-import NetworkMap.Certificates.Model exposing (Model)
+import Array exposing (Array)
+import Dict exposing (Dict)
+import List
+import List.FlatMap
+import List.Unique
+import NetworkMap.Certificates.Model exposing (Certificate, CertificateIds, Model, Trust)
 import NetworkMap.Certificates.Msg exposing (Msg(..))
 import NetworkMap.Certificates.Port as Port
+import Utils.ArrayExtras as ArrayExtras
+
+
+convertCert : Certificate -> List ( String, String, Trust )
+convertCert cert =
+    let
+        chain =
+            cert.chain
+
+        ( _, pairs ) =
+            chain |> Array.foldl gatherPairs ( Nothing, [] )
+    in
+    pairs
+
+
+gatherPairs : Trust -> ( Maybe String, List ( String, String, Trust ) ) -> ( Maybe String, List ( String, String, Trust ) )
```

Comment:
Maybe define a type for `(String, String, Trust)`?

`gather` doesn't give any insight into what happens there.